db/background_mgr_attachment_compaction.go

```diff
@@ -100,6 +103,25 @@ func (a *AttachmentCompactionManager) Init(ctx context.Context, options map[stri
 	return newRunInit()
 }
 
+func (a *AttachmentCompactionManager) PurgeDCPMetadata(ctx context.Context, datastore base.DataStore, database *Database, phase string) error {
```

Comment:
can we have this just take a DCPClient which you'd return from `attachmentCompactMarkPhase` etc?

I'm trying to make this easier to read. If this doesn't work, that's okay, but we could call a function without having to worry about reconstructing the prefix that seems preferable.

Reply:
I've addressed this issue by extracting the key prefix from the client in the code when the client is created. Then returning that in the phases that we need to handle rollback on and using that to pass into this function.